scripts/Tools/case.setup

```diff
@@ -56,9 +67,14 @@ def _main_func(description):
         test_results = doctest.testmod(verbose=True)
         sys.exit(1 if test_results.failed > 0 else 0)
 
-    caseroot, clean, test_mode, reset = parse_command_line(sys.argv, description)
-    with Case(caseroot, read_only=False) as case:
-        case_setup(case, clean=clean, test_mode=test_mode, reset=reset)
+    caseroot, clean, test_mode, reset, usermods_dir = parse_command_line(sys.argv, description)
+
+    if usermods_dir is not None:
+        logger.warn("WARNING: REMOVING all user_nl_xxx files as part of ./case_setup --apply-usermods")
+        apply_user_mods(caseroot, usermods_dir, cleanfirst=True)
```

Comment:
I thought we avoid having any substantive logic or behavior in the top-level scripts, instead putting it in the associated function (case_setup in this case).